iio/iio_demo/iio_demo.c

```diff
@@ -0,0 +1,420 @@
+ * @param channel - Channel properties.
+ * @return: Number of bytes written to device, or negative value on failure.
+ */
+static ssize_t set_demo_attr(void *device, char *buf, size_t len,
+			     const struct iio_ch_info *channel)
+{
+	ssize_t ret = SUCCESS;
+
+	demo_par = srt_to_uint32(buf);
+
+	if (ret < 0)
+		return ret;
+
+	return len;
+}
+
+static struct iio_attribute iio_attr_demo = {
+	.name = "demo_attr",
+	.show = get_demo_attr,
+	.store = set_demo_attr,
+};
+
+static struct iio_attribute *demo_attributes[] = {
+	&iio_attr_demo,
+	NULL,
+};
+
+static struct iio_channel iio_demo_channel_voltage0_in = {
+	.name = "voltage0",
+	.attributes = demo_attributes,
+	.ch_out = false,
+};
+
+static struct iio_channel *iio_demo_channels[] = {
+	&iio_demo_channel_voltage0_in,
+	NULL,
+};
+
+static struct iio_attribute *iio_demo_global_attributes[] = {
+	&iio_attr_demo,
+	NULL,
+};
+
+/**
+ * @brief Create structure describing a device, channels and attributes.
+ * @param device_name - Device name.
+ * @param num_ch - Number of channels that the device has.
+ * @return iio_device or NULL, in case of failure.
+ */
+static struct iio_device *iio_demo_create_device(const char *device_name,
+		uint16_t num_ch)
+{
+	struct iio_device *iio_device;
+
+	iio_device = calloc(1, sizeof(*iio_device));
+	if (!iio_device)
+		return NULL;
+
+	iio_device->name = device_name;
+	iio_device->num_ch = num_ch;
+	iio_device->attributes = iio_demo_global_attributes;
+	iio_device->channels = iio_demo_channels;
+
+	return iio_device;
+}
+
+/**
+ * @brief Delete iio_device.
+ * @param iio_device - Structure describing a device, channels and attributes.
+ * @return SUCCESS in case of success or negative value otherwise.
+ */
+static ssize_t iio_demo_delete_device(struct iio_device *iio_device)
+{
+	if(iio_device)
+		free(iio_device);
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Transfer data from RAM to device.
+ * @param iio_inst - Physical instance of a iio_demo_dac device.
+ * @param bytes_count - Number of bytes to transfer.
+ * @param ch_mask - Opened channels mask.
+ * @return Number of bytes transfered, or negative value in case of failure.
+ */
+static ssize_t iio_demo_transfer_mem_to_dev(void *iio_inst,
+		size_t bytes_count,
+		uint32_t ch_mask)
+{
+	struct iio_demo_device *demo_device;
+	demo_device = (struct iio_demo_device *)iio_inst;
+	if (!demo_device)
+		return FAILURE;
+
+	return bytes_count;
+}
+
+/**
+ * @brief Transfer data from device into RAM.
+ * @param iio_inst - Physical instance of a device.
+ * @param bytes_count - Number of bytes to transfer.
+ * @param ch_mask - Opened channels mask.
+ * @return bytes_count or negative value in case of error.
+ */
+static ssize_t iio_demo_transfer_dev_to_mem(void *iio_inst,
+		size_t bytes_count,
+		uint32_t ch_mask)
+{
+	struct iio_demo_device *demo_device;
+	demo_device = (struct iio_demo_device *)iio_inst;
+	if (!demo_device)
+		return FAILURE;
+
+	/* Implement if necessary. */
```

Comment:
nitpick: I don't understand the comment. Maybe it could be a little more explicit on what should be implemented here.